src/docs/product/issues/issue-details/performance-issues/n-one-api-calls.mdx

```diff
@@ -0,0 +1,64 @@
+---
+title: "N+1 API Calls"
+sidebar_order: 15
+description: "Learn more about N+1 API calls and how to diagnose and fix them."
+---
+
+<Include name="early-adopter-note.mdx" />
+
+_N+1 API Calls_ are a performance problem in which an application makes many simultaneous network requests to fetch a given type of resource. For example, an application might make one network request for each item in a list, instead of fetching all the information in a combined network call. Each call to the server incurs some performance overhead, and if the number of calls is high this can lead to severe performance problems. This problems commonly occurs when rendering lists or grids of items, where each item needs to fetch additional information from the server.
+
+## Detection Criteria
+
+The N+1 API Calls detector looks for groups of simultaneous network calls to the same URL. It uses has the following criteria for detection:
+
+- applies to `GET` requests only
+- a minimum of 10 network calls need to happen at the same time
+- the network calls must be a minimum of 50s in duration, and within 5ms of each other
+- the calls must not be to a GraphQL endpoint, or to `_next/data`
+- the calls must not be fetching a static resource (e.g., `.png` files)
+
+## Span Evidence
+
+The evidence for an N+1 queries problem has three main aspects:
+
+- Transaction name
+- Repeating spans - This is the URL that's receiving the repeated network calls
+- Parameters - These are the query parameters that vary between the network calls
+
+![N+1 API Calls span evidence](n-plus-one-api-calls-evidence.png)
+
+## Fingerprinting
+
+Sentry computes the issue fingerprint based on the URLs of the repeating network calls. Sentry parameterizes the URLs by stripping integers, UUIDs, SHA1 hashes, and MD5 hashes. If you're finding that Sentry is failing to correctly group N+1 API Calls issues, you can parameterize the `http.client` spans in your transactions by using the <PlatformLink to="/configuration/options#before-send-transaction"><PlatformIdentifier name="before-send-transaction" /></PlatformLink> hook if your platform supports it.
+
```

Comment:
```suggestion
Sentry computes the issue fingerprint based on the URLs of the repeating network calls. Sentry parameterizes the URLs by stripping integers, UUIDs, SHA1 hashes, and MD5 hashes. If you're finding that Sentry is failing to correctly group N+1 API Call issues, you can parameterize the `http.client` spans in your transactions by using the <PlatformLink to="/configuration/options#before-send-transaction"><PlatformIdentifier name="before-send-transaction" /></PlatformLink> hook if your platform supports it.

```

Reply:
Do you have any advice on how to correctly use "N+1 API Calls" vs. "N+1 API Call"? There are opinions about this flying around, and it's not very consistent. e.g., in the Sentry app the page title is "N+1 Query" and "N+1 API Call" on the issue details page, but the documentation often says "N+1 Queries" and "N+1 API Calls". Is there a good way to figure out what to use and when? Should it always be consistent?

Reply:
I think we should be consistent with the Sentry app whenever possible. So if the app calls it "query" and "call" singular, we should refer to it as singular in docs too, at least when we're referring to the feature name. When we're not referring to it as the feature name, (like the sentence: If you're finding that Sentry is failing to correctly group N+1 API Call issues), then we should just go with what's grammatically correct :)

Reply:
Fair enough! So, should this page be called "N+1 API Query", then? And similarly for "N+1 Query"?

Reply:
Updated the pluralization so it says "N+1 API Call" whenever it explicitly mentions the detector!